modules/installing-gcp-user-defined-labels-and-tags.adoc

```diff
@@ -0,0 +1,123 @@
+
+[NOTE]
+====
+Labels & tags once defined during the install-config creation phase cannot be modified or updated with new labels & tags after cluster creation.
+====
+
+Support for user-defined labels & tags for GCP resources is available in {product-title} version {product-version} as Technology Preview (TP).
+
+Service Account used for cluster creation must have the `TagUser` roles assigned to it, required for tags defined at both organization and project level.
+
+You can use `.platform.gcp.userLabels` and `platform.gcp.userTags` fields to define the list of desired labels & tags in `install-config.yaml`. `.platform.gcp.userLabels` and `.platform.gcp.userTags` fields are a yaml array of objects type.
+
+.Sample `install-config.yaml` file
+[source,yaml]
+----
+apiVersion: v1
+featureSet: TechPreviewNoUpgrade
+platform:
+ gcp:
+   userLabels:
+   - key: <label_key>
+     value: <label_value>
+   userTags:
+   - parentID: <OrganizationID/ProjectID>
+     key: <tag_key_short_name>
+     value: <tag_value_short_name>
+----
+
+The user-defined labels have the following requirements:
+
+* A label key and value must have a minimum of 1 character and can have a maximum of 63 characters.
+* A label key and value must contain only lowercase letters, numeric characters, underscores, and dashes (i.e. _-).
+* A label key must start with a lowercase letter.
+* You can configure a maximum of 32 labels per resource.
+and each resource can have a maximum of 64 labels, and 32 labels are reserved for internal use by {product-title}.
+
+The user-defined tags have the following requirements:
+
+* Tag Keys and Tag Values must already exist and will not be created by {product-title}.
+* A tag `parentID` can be either `OrganizationID` or `ProjectID`.
+** An `OrganizationID` must consist of decimal numbers, and cannot have leading zeros.
+** A `ProjectID` must be 6 to 30 characters in length, can only contain lowercase letters, numbers, and hyphens, and must start with a letter, and cannot end with a hyphen.
+* A tag key and value must have a minimum of 1 character and can have a maximum of 63 characters.
+* A tag key must contain only uppercase and lowercase alphanumeric characters, hyphens, underscores, and periods (i.e. ._-).
+* A tag value must contain only uppercase and lowercase alphanumeric characters, hyphens, underscores, periods, at signs, percent signs, equals signs, plusses, colons, commas, asterisks, pound signs, ampersands, parentheses, square braces, curly braces, and spaces (i.e. _-.@%=+:,*#&(){}[] and spaces).
+* A tag key and value must begin and end with an alphanumeric character.
+* Tag value is of type scalar and must be one of the pre-defined values for the key.
+* You can configure a maximum of 50 tags.
+* There should be no tag key with the same value as any of the existing tag keys on a resource(inherited from parent resource).
+
+For more information about identifying the `OrganizationID` or `ProjectID`, see:
```

Comment:
Move these links to an additional resources section? In the preview, I see floating asterisks.

The lists and the links just seem dropped in among the surrounding conceptual information. 

![Screenshot from 2023-08-28 16-22-33](https://github.com/openshift/openshift-docs/assets/57954076/0a4781fa-42df-4d96-b89d-3f9f47774a99)

Reply:
All of these requirements and restrictions, are they defined by Google or Red Hat? If Google, how will these lists be maintained? Would it make sense to refer users to the Google docs for these requirements and restrictions?

Reply:
There are many instances, that I have come across, where writers have provided links outside of Redhat. Let me get this confirmed once again.

Reply:
Yeah, this is a tough area and installation docs often refer to vendor documentation as they own the platform and its capabilities.

Reply:
Subheadings are great and really divide the information. Nice!